### PR TITLE
Plugins: remove deprecated code (state handling)

### PR DESCRIPTION
**Pull request by leventebalogh**
**Related issue:** https://github.com/grafana/grafana/issues/40503

**⚠️ &nbsp; This branch is depending on https://github.com/grafana/grafana/tree/leventebalogh/remove-deprecated-plugin-components.**

### What changed
- [x] added a new selector and hook for getting plugin errors (used by `<PluginErrorsInfo />`)
- [x] removed references to the old (deprecated) state handling and used the new one (plugins/admin/state)
- [x] removed old state handling code

### CODE REVIEW

File: public/app/features/plugins/components/PluginsErrorsInfo.tsx [hunk 1]
```diff
@@ -1,42 +1,19 @@
...
-}
-const connector = connect(mapStateToProps, mapDispatchToProps);
-type PluginsErrorsInfoProps = ConnectedProps<typeof connector> & OwnProps;
-
-export const PluginsErrorsInfoUnconnected: React.FC<PluginsErrorsInfoProps> = ({
-  loadPluginsErrors,
-  errors,
-  children,
-}) => {
+export const PluginsErrorsInfo: React.FC = ({ children }) => {
```
Comment (jackw) [inline public/app/features/plugins/components/PluginsErrorsInfo.tsx:8]: nit: I think we prefer functional components to be defined like the following these days:

```suggestion
export function PluginsErrorsInfo({ children }): React.ReactElement | null {
```

**Review by jackw — APPROVED** [review]
Tested locally and everything appears to be working as expected. Excellent work @leventebalogh ! 🏅

File: public/app/features/plugins/components/PluginsErrorsInfo.tsx (hunk 1 above)
Comment (leventebalogh) [reply public/app/features/plugins/components/PluginsErrorsInfo.tsx:8]: Absolutely, will change 👍

Comment (leventebalogh) [reply public/app/features/plugins/components/PluginsErrorsInfo.tsx]: Changed in https://github.com/grafana/grafana/pull/41739/commits/ff287df5b61fe0d7eb7e20380a77af0a3791f7ab.

**Pull request merged**